### DISCUSSION TITLE
Error when build ios

### DISCUSSION THREAD

**Discussion by lichongting** [maplibre/maplibre-native#1962] [Q&A]
I'm new to ios and when I compile platform/ios as per re ad me, I get the following error: 
`lichongting@lichongtingdeMac ios % bazel run //platform/ios:App
Starting local Bazel server and connecting to it...
INFO: Analyzed target //platform/ios:App (72 packages loaded, 6350 targets configured).
INFO: Found 1 target...
ERROR: /Users/lichongting/Workspace/maplibre-native/BUILD.bazel:119:8: Executing genrule //:git_hash failed: (Exit 127): bash failed: error executing command (from target //:git_hash) /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; 
        git rev-parse HEAD > bazel-out/ios-x86_64-min12.0-applebin_ios-ios_x86_64-fastbuild-ST-41567f1bc2e5/bin/hash
    '
: command not found
Target //platform/ios:App failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 23.384s, Critical Path: 0.84s
INFO: 9 processes: 9 internal.
FAILED: Build did NOT complete successfully
ERROR: Build failed. Not running target`
Can someone help me or explain the reason for this error? thanks

**Comment by louwers** [comment]
hi @lichongting Do you have git installed?

**Reply by lichongting** [reply]
Thanks for the reply. I've confirmed that Git is installed. Eventually, I found out that it might be due to the encoding format, because I deleted the error line and rewrote it again, it didn't report the error again
`genrule(
    name = "git_hash",
    outs = ["hash"],
    cmd = """git rev-parse HEAD > $@""",
    local = True,
    visibility = ["//visibility:public"],
)`

**Reply by louwers** [reply]
Thanks for reporting back, this may help someone else with the same issue.
